packages/odata-v2/src/de-serializers/temporal-de-serializers.ts

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable valid-jsdoc */
+import { Temporal } from '@js-temporal/polyfill';
+import { DeSerializer } from '@sap-cloud-sdk/odata-common/internal';
+
+/**
+ * Temporal (de-)serializers for Odata-v2.
+ * @internal
+ */
+export const temporalDeSerializers = {
+  'Edm.DateTime': {
+    deserialize: deserializeToPlainDateTime,
+    serialize: serializeFromPlainDateTime,
+    serializeToUri: (value, serialize) =>
+      `datetime'${serialize(value).toString().replace(/Z$/, '')}'`
+  } as DeSerializer<Temporal.PlainDateTime>,
+  'Edm.DateTimeOffset': {
+    deserialize: deserializeToZonedDateTime,
+    serialize: serializeFromZonedDateTime,
+    serializeToUri: (value, serialize) => `datetimeoffset'${serialize(value)}'`
+  } as DeSerializer<Temporal.ZonedDateTime>,
+  'Edm.Time': {
+    deserialize: deserializeToPlainTime,
+    serialize: serializeFromPlainTime,
+    serializeToUri: (value, serialize) => `time'${serialize(value)}'`
+  } as DeSerializer<Temporal.PlainTime>
+};
+
+/**
+ * @internal
+ */
+export function deserializeToPlainDateTime(
+  edmDateTime: string
+): Temporal.PlainDateTime {
+  const dateTimeOffsetComponents = parseEdmDateTimeOffset(edmDateTime);
+  const dateTimeInstant = Temporal.Instant.fromEpochMilliseconds(
+    parseInt(dateTimeOffsetComponents.ticks)
+  );
+  return Temporal.PlainDateTime.from(dateTimeInstant.toString());
+}
+
+/**
+ * @internal
+ */
+export function deserializeToZonedDateTime(
+  edmDateTime: string
+): Temporal.ZonedDateTime {
+  const dateTimeOffsetComponents = parseEdmDateTimeOffset(edmDateTime);
+  const dateTimeInstant = Temporal.Instant.fromEpochMilliseconds(
+    parseInt(dateTimeOffsetComponents.ticks)
+  );
+  if (dateTimeOffsetComponents.sign && dateTimeOffsetComponents.offset) {
+    const timeZone = Temporal.TimeZone.from(
+      dateTimeOffsetComponents.sign + dateTimeOffsetComponents.offset
+    );
+    return dateTimeInstant.toZonedDateTimeISO(timeZone);
+  }
+  return dateTimeInstant.toZonedDateTimeISO('UTC');
+}
+
+function parseEdmDateTimeOffset(edmDateTime: string): {
+  [key: string]: string;
+} {
+  const dateTimeOffsetComponents =
+    /^\/Date\((?<ticks>\d+)((?<sign>[+-])(?<offset>\d{4}))?\)\/$/.exec(
+      edmDateTime
+    )?.groups;
+  if (!dateTimeOffsetComponents) {
+    throw new Error(`Failed to parse edmDateTime: ${edmDateTime}.`);
+  }
+  return dateTimeOffsetComponents;
+}
+
+/**
+ * @internal
+ */
+export function serializeFromPlainDateTime(
+  value: Temporal.PlainDateTime
+): string {
+  const instant = Temporal.TimeZone.from('UTC').getInstantFor(value);
+  return `/Date(${instant.epochMilliseconds})/`;
+}
+
+/**
+ * @internal
+ */
+export function serializeFromZonedDateTime(
+  value: Temporal.ZonedDateTime
+): string {
+  const operator = value.offset[0];
+  const offset = value.offset.replace(/[-+:]/g, '');
+  return `/Date(${value.epochMilliseconds}${operator}${leftpad(offset, 4)})/`;
```

Comment:
The format looks like `+00:00`  which is replaced to `0000` by removing the sign and :. So when is the length shorter 4 and the leftpad is needed?